### PR TITLE
EPMDJ-7722 Improve performance admin-part

### PR DESCRIPTION
**Pull request by LikhanovMaxim**
optimize toCoverMap

### CODE REVIEW

File: admin-part/src/main/kotlin/Methods.kt [hunk 1]
```diff
@@ -21,12 +21,14 @@ import com.epam.drill.plugins.test2code.util.*
 import kotlinx.serialization.*
 
 @Serializable
-internal data class Method(
+data class Method(
```
Comment (LikhanovMaxim) [inline admin-part/src/main/kotlin/Methods.kt:24]: why did you remove?

File: admin-part/src/main/kotlin/Plugin.kt [hunk 2]
```diff
@@ -34,7 +34,7 @@ import kotlinx.serialization.json.*
 import java.io.*
 import java.util.concurrent.*
 
-internal object AsyncJobDispatcher : CoroutineScope {
+object AsyncJobDispatcher : CoroutineScope {
```
Comment (LikhanovMaxim) [inline admin-part/src/main/kotlin/Plugin.kt:37]: why did you remove?

File: admin-part/src/main/kotlin/coverage/Ext.kt [hunk 3]
```diff
@@ -87,22 +88,17 @@ internal fun BundleCounter.toCoverDto(
...
-        }
-    }.flatMap { it.await() }.toMap()
+): Map<Method, CoverMethod> = bundle.packages.let { packages ->
+    val map = packages.parallelStream().flatMap { it.classes.stream() }.flatMap { c ->
+        c.methods.stream().map { m -> m.key to m }
+    }.collect(Collectors.toMap({ it.first }, { it.second }))
+    parallelStream().map { method ->
+        val covered = method.toCovered(map[method.key])
+        covered.takeIf { !onlyCovered || it.count.covered > 0 }?.let { method to it }
+    }.filter { it != null }.collect(Collectors.toMap({ it!!.first }, { it!!.second }))
```
Comment (LikhanovMaxim) [inline admin-part/src/main/kotlin/coverage/Ext.kt:101]: !! it can be NPE
it would be better to check like this: it?.first != null

File: admin-part/src/main/kotlin/coverage/Ext.kt [hunk 4]
```diff
@@ -153,7 +147,7 @@ internal fun Method.toCovered(count: Count?) = CoverMethod(
     coverageRate = count?.coverageRate() ?: CoverageRate.MISSED
 )
 
-internal fun Method.toCovered(counter: MethodCounter? = null): CoverMethod = toCovered(counter?.count)
+fun Method.toCovered(counter: MethodCounter? = null): CoverMethod = toCovered(counter?.count)
```
Comment (LikhanovMaxim) [inline admin-part/src/main/kotlin/coverage/Ext.kt:150]: why did you remove it?

File: admin-part/src/main/kotlin/coverage/Model.kt [hunk 5]
```diff
@@ -21,7 +21,7 @@ import com.epam.drill.plugins.test2code.util.*
 import com.epam.kodux.util.*
 import kotlinx.serialization.*
 
-internal data class DiffMethods(
+data class DiffMethods(
```
Comment (LikhanovMaxim) [inline admin-part/src/main/kotlin/coverage/Model.kt:24]: why did you remove it?

File: admin-part/src/main/kotlin/coverage/Model.kt [hunk 6]
```diff
@@ -112,16 +116,15 @@ data class ClassCounter(
...
+    val fullName: String,
+) : NamedCounter(), JvmSerializable
 
 @Serializable
 data class MethodCounter(
     override val name: String,
     val desc: String,
     val decl: String,
+    val sign: String,
+    val key: String,
```
Comment (LikhanovMaxim) [inline admin-part/src/main/kotlin/coverage/Model.kt:128]:
```suggestion
    val fullName: String,
```

File: admin-part/src/main/kotlin/jvm/Ext.kt [hunk 7]
```diff
@@ -25,10 +25,10 @@ import java.io.*
 internal fun ClassCounter.parseMethods(classBytes: ByteArray): List<Method> = run {
     val classParser = ClassParser(ByteArrayInputStream(classBytes), fullName)
     val parsedMethods = classParser.parse().run {
-        methods.associateBy { it.name to it.signature }
+        methods.associateBy { "${it.name}${it.signature}".weakIntern() }
     }
     methods.map { m ->
-        val method = parsedMethods[m.name to m.desc]
+        val method = parsedMethods[m.sign.weakIntern()]//todo check is it faster?
```
Comment (LikhanovMaxim) [inline admin-part/src/main/kotlin/jvm/Ext.kt:31]: it is faster. you can remove this todo

**Review by IgorKey — APPROVED** [review]

File: admin-part/src/main/kotlin/jvm/JacocoExt.kt [hunk 8]
```diff
@@ -181,3 +184,18 @@ fun parseDescType(char: Char, charIterator: CharIterator): String = when (char)
 private fun ExecClassData.toExecutionData(probeIds: Map<String, Long>): ExecutionData? = probeIds[className]?.let {
     ExecutionData(it, className, probes.toBooleanArray())
 }
+
+private fun Analyzer.fixConcurrencyIssues() = also { analyzer ->
```
Comment (LikhanovMaxim) [inline admin-part/src/main/kotlin/jvm/JacocoExt.kt:188]: rename or creating a new object

File: admin-part/src/main/kotlin/ScopeHandler.kt [hunk 9]
```diff
@@ -44,6 +46,19 @@ internal fun Plugin.initActiveScope(): Boolean = activeScope.init { sessionChang
...
 
+fun Plugin.initSessionHandler(): ActiveSessionHandler = { map ->
+    val context = state.coverContext()
+    val bytes = classBytes.value ?: adminData.loadClassBytes().also { bytes -> classBytes.updateAndGet { bytes } }
+    val preparedBundle = map.keys.associateWithTo(mutableMapOf()) {
+        BundleCounter.empty
+    }
+    val calculated = map.mapValuesTo(preparedBundle) {
+        it.value.bundle(context, bytes)
+    }
```
Comment (LikhanovMaxim) [inline admin-part/src/main/kotlin/ScopeHandler.kt:57]: refactoring

File: admin-part/src/main/kotlin/Session.kt [hunk 10]
```diff
@@ -77,8 +86,24 @@ class ActiveSession(
                 }?.let { map.put(typedTest, it) } ?: map
             }
         }
+    }.also {
+        channel.send(dataPart.asSequence().map { it.testName.weakIntern().typedTest(testType.weakIntern()) }.distinct())
     }
 
+    private val channel = Channel<Sequence<TypedTest>>()
```
Comment (LikhanovMaxim) [inline admin-part/src/main/kotlin/Session.kt:92]: rename

File: admin-part/src/main/kotlin/Session.kt [hunk 11]
```diff
@@ -115,6 +145,9 @@ data class FinishedSession(
     override val tests: Set<TypedTest>,
     override val testStats: Map<TypedTest, TestStats> = emptyMap(),
     val probes: List<ExecClassData>,
+    @Transient
+    @kotlin.jvm.Transient
```
Comment (LikhanovMaxim) [inline admin-part/src/main/kotlin/Session.kt:148]: choose one
